OOP/RationalNumber.cs

```diff
@@ -0,0 +1,151 @@
+﻿
+namespace OOP
+{
+    public struct RationalNumber
+    {
+        /// <summary>
+        /// Поля
+        /// </summary>
+        private int _numerator;
+        private int _denominator;
+        private float Rational=>((float)_numerator)/((float)_denominator);
+        /// <summary>
+        /// Свойства
+        /// </summary>
+        public int Numerator { get => _numerator; }
+       
+        public int Denominator { get => _denominator; }
+
+        /// <summary>
+        /// Конструктор
+        /// </summary>
+        /// <param name="Numerator">Числитель</param>
+        /// <param name="Denominator">Знаменатель</param>
+        /// <exception cref="ArgumentException">исключение</exception>
+        public RationalNumber(int Numerator, int Denominator)
+        {
+            if (Denominator == 0)
+                throw new ArgumentException("Делитель не может быть равен нулю");
+
+            if (Denominator < 0)
+            {
+                Numerator *= -1;
+                Denominator *= -1;
```

Comment:
```suggestion
                Numerator = -Numerator;
                Denominator = -Denominator;
```
Так будет немного быстрее